---
title: "scTCR-seq"
subtitle: ""
date: "`r format(Sys.Date())`"
output:  
  rmdformats::robobook: 
    code_folding: show 
    number_sections: FALSE
    toc_depth: 6
    toc_float: false
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=T, fig.align = "center", 
                      message=F, warning=F,
                      results = "markup",
                      error = TRUE,
                      highlight = TRUE,
                      prompt = FALSE,
                      tidy = FALSE)
```


```{r, echo=FALSE}
library(dplyr)
library(ggplot2)
# library(DESeq2)
# library(edgeR)
library(reshape)
library(dplyr)
```


<a href="https://sung2021.github.io/">Back to Main Page </a>  

# Workflow 

## Preprocessing by cellranger (count, vdj, aggr)  

```{bash, echo=T, eval=FALSE}
# Sample : Control
# CellRanger count for Transcriptome 
cellranger count --id=<SampleID_GEX> \
                 --transcriptome=<Path_to_Mouse_GEX_Reference> \
                 --fastqs=<Path_to_GEX_Raw_Data> \
                 --sample=<Sample_Name_GEX> \
                 --expect-cells=10000

# CellRanger vdj for VDJ information  
cellranger vdj --id=<SampleID_VDJ> \
               --reference=<Path_to_Mouse_VDJ_Reference> \
               --fastqs=<Path_to_VDJ_Raw_Data> \
               --sample=<Sample_Name_VDJ>

  
# Sample : Treated
# CellRanger count for Transcriptome 
cellranger count --id=<SampleID_GEX_2> \
                 --transcriptome=<Path_to_Mouse_GEX_Reference> \
                 --fastqs=<Path_to_GEX_Raw_Data_2> \
                 --sample=<Sample_Name_GEX_2> \
                 --expect-cells=10000

# CellRanger vdj for VDJ information  
cellranger vdj --id=<SampleID_VDJ_2> \
               --reference=<Path_to_Mouse_VDJ_Reference> \
               --fastqs=<Path_to_VDJ_Raw_Data_2> \
               --sample=<Sample_Name_VDJ_2>

# CellRanger aggr for combining all samples
cellranger aggr --id=<Aggregation_ID> \
                --csv=<Path_to_Aggregation_CSV>
```

[Cell Ranger 5' VDJ Algorithm Overview](https://www.10xgenomics.com/support/software/cell-ranger/latest/algorithms-overview/cr-5p-vdj-algorithm)


<br>

## VDJ information     

```{r, eval=FALSE, echo=TRUE}
# Import cellranger aggr output 
vdj <- read.csv('path_to_filtered_contig_annotations.csv')
clono <- read.csv('path_to_clonotypes.csv')

# Removal of duplicates and merging by clonotype_id
vdj <- vdj %>% distinct(barcode, .keep_all = TRUE)
rownames(vdj) <- vdj$barcode
vdj.clono <- inner_join(vdj, clono, by=c('raw_clonotype_id'='clonotype_id'))

# Counting TRA and TRB from cdr3s_aa  
vdj.clono$TRBcount <- stringr::str_count(vdj.clono$cdr3s_aa, "TRB")
vdj.clono$TRAcount <- stringr::str_count(vdj.clono$cdr3s_aa, "TRA")
vdj.clono <- vdj.clono %>% arrange(desc(frequency))
rownames(vdj.clono) <- vdj.clono$barcode

# filtering 1 alpha chain and  1 beta chain clonotype
vdj.clono <- vdj.clono %>% filter(TRBcount == 1 & TRAcount == 1)

# save vdj information
vdj.clono %>% write.csv('path_to_save_vdj_clono.csv')

```

<br>

## Basic Clonotype Analysis   

**Immune repertoire analysis**  

Analysis using R packages (scRepertoire, immunarch)  


**1. Diversity Index** 

  - Shannon Entropy 
  - Simpson 

**2. Repertoire analysis** 

  - TRAV/D/J/C distribution
  - TRBV/D/J/C distribution
  - CDR3nt length distribution
  - Clustering of clonotype repertoire

**3. Expansion study**  

  - Clonotype expansion (hyper,high,medium,small,rare)
  - Top 100 expanded clonotypes with TRAV:TRBV info

<br>

## Further Analysis with scRNA-seq data     

Merging vdj information with scRNA-seq (if available)  
Suggested analyses below  


**Target cell selection**  

  - Cells with target features (e.g. cytotoxic cells, effector T cells, NKT, etc). 
  - Cells associated with expanded clonotypes (hyper, high, medium, small, rare). 
  - Cells with clonotypes displaying treatment-specific response.  

**Correlation of clonotype frequency with related data**  

  - Correlation of clonotype frequency with target gene expression profiles.  
  - Correlation with clinical outcomes and phenotypic data. (if available)  

**Comparative Analysis**  

  - Comparison between different patient groups (e.g., responders vs non-responders to therapy, different disease stages) and associated clonotypes.
  - Clonotype repertoires to identify conserved features of the immune response.

**Statistical Modeling**  

  - Application of statistical models and computational algorithms for the analysis.  
  - Evaluation of the robustness and reproducibility of findings.  


<br><br>

 
<span style="font-size: 34px;">**Example Plots**</span>

# Clonotype Distribution 
```{r, echo=FALSE}
# 데이터 프레임 예제 데이터 생성
set.seed(123) # 결과의 재현 가능성을 위해 설정
clonotypes <- paste0("Clonotype", 1:100) # 100개의 클론타입 예제
A_values <- sample(10:100, 100, replace = TRUE) # A 값 예제 데이터
B_values <- sample(1:100, 100, replace = TRUE) # B 값 예제 데이터

# 데이터 프레임 생성
data <- data.frame(Clonotype = clonotypes, A = A_values, B = B_values)

# 긴 형태로 데이터를 변환
data_melted <- data %>% reshape2::melt(id.vars = "Clonotype")
```

```{r, echo=TRUE, fig.width=12, fig.height=5}
# data_melted : Clonotype frequency information 
# Draw plot
ggplot(data_melted, aes(x = reorder(Clonotype, -value), y = value, 
                        fill = variable)) + 
  geom_col(color = "grey3", size=0.2) + 
  scale_fill_manual(values = c("salmon", "darkorange"),
                    labels = c("Sample A", "Sample B")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, size=6),
        legend.title = element_blank(),
        plot.title = element_text(hjust = 0.5, size=20)) +
  labs(x = "Top 100 Clonotype", y = "Total number of clonotype",
      title = "Clonotype distribution by sample") 
```

<br>


# Clonotype Abundance in each sample {.tabset}  

```{r, echo=FALSE}
library(packcircles)
```

```{r, echo=TRUE}
# Bubble Plot 
# input data format
# column1 : name of the factor
# column2 : frequency of the factor

bubble_plot = function(input.data, title="Abundant Clonotypes"){
  colnames(input.data)[2] = "Freq"
  
  packing <- circleProgressiveLayout(input.data$Freq, sizetype='area')
  
  input.data <- cbind(input.data, packing)
  colnames(input.data)[1] ='clonotype'
  dat.gg <- circleLayoutVertices(packing, npoints=50)
  
  ggplot() + 
    # Make the bubbles
    geom_polygon(data = dat.gg, aes(x, y, group = id, 
                                    fill=as.factor(id)), 
                 colour = "black", alpha = 0.6) +
    theme_void() + 
    theme(legend.position="none",
          plot.title = element_text(hjust = 0.5, size=30)) +
    coord_equal() +
    # Add text in the center of each bubble + control its size
    geom_text(data = input.data, aes(x, y, size=Freq, label = clonotype)) +
    scale_size_continuous(range = c(0.5,4)) +
    ggtitle(title) +
    scale_fill_viridis_d(option = "D", begin = 0.0, end = 1.0, direction = 1) 
}
```


## Sample A  
```{r,fig.width=15, fig.height=9}
bubble_plot(input.data = data[,c(1,2)], title = "Abundant Clonotypes in A")
```


## Sample B  
```{r,fig.width=15, fig.height=9}
bubble_plot(input.data = data[,c(1,3)], title = "Abundant Clonotypes in B")
```

<br><br>

# Gene Expression Profiling across Clonotypes and Samples   
```{r, echo=FALSE}
library(MASS)
set.seed(123)
# 셀 수와 유전자 목록 설정
n_cells <- 1000
genes <- c("NKG7", "KLRG1", "INFG", "GZMA", "PRF1")

# 공분산 행렬 정의 - NKG7과 KLRG1은 높은 상관관계를, 나머지는 낮은 상관관계를 가지게 됩니다.
# 공분산 행렬은 각 유전자 간의 상관관계를 정의합니다. 이 경우에는 상관관계가 다음과 같습니다:
#   NKG7과 KLRG1: 0.8
#   INFG, GZMA, PRF1 간의 상관관계: 0.4
#   나머지 상관관계는 0.2로 설정
cov_matrix <- matrix(c(1,    0.8,  0.2,  0.2,  0.2,
                       0.8,  1,    0.2,  0.2,  0.2,
                       0.2,  0.2,  1,    0.4,  0.4,
                       0.2,  0.2,  0.4,  1,    0.4,
                       0.2,  0.2,  0.4,  0.4,  1), byrow = TRUE, nrow = 5)

# 각 유전자의 평균 발현값
means <- rep(10, length(genes))

# 공분산 행렬과 평균값을 사용하여 연관된 정규분포 발현 데이터를 생성
expression_matrix <- mvrnorm(n = n_cells, mu = means, Sigma = cov_matrix, empirical = TRUE)

# 발현 데이터를 행렬 형태로 변환하고, 행렬에 적절한 이름을 지정합니다.
expression_matrix <- matrix(expression_matrix, nrow = n_cells, ncol = length(genes),
                            dimnames = list(paste0("cell", 1:n_cells), genes))

# 각 셀에 대한 샘플 정보와 클론타입 정보를 가진 데이터 프레임 생성
set.seed(123) # 결과의 재현성을 위한 시드 설정

# 클론타입에 가중치를 주어 샘플링
clonotype_weights <- rep(1, 25)  # 모든 클론타입에 대해 기본 가중치 설정
clonotype_weights[c(1, 25)] <- c(10, 0.1)  # 클론타입1은 더 자주, 클론타입25는 더 드물게

cell_metadata <- data.frame(
  cell_id = paste0("cell", 1:n_cells),
  sample = sample(paste0("Condition", c("A", "B", "C")), n_cells, replace = TRUE),
  clonotype = sample(paste0("clonotype", 1:25), n_cells, replace = TRUE, prob = clonotype_weights)
)
rownames(cell_metadata) = cell_metadata$cell_id
cell_metadata = cell_metadata[,-1]

clonotype_freq <- cell_metadata %>%
  group_by(clonotype) %>%
  summarise(frequency = n()) %>%
  arrange(desc(frequency))

clonotype_order = clonotype_freq$clonotype

cell_metadata$clonotype = factor(cell_metadata$clonotype, levels = clonotype_order) 
cell_metadata = cell_metadata %>% arrange(clonotype)

```
<br>

## Heatmap (Cytotoxic Genes)
```{r, fig.width=8, fig.height=9, out.width="70%", echo=TRUE}
# Draw heatmap 
expression_matrix[rownames(cell_metadata),] %>% 
  pheatmap::pheatmap(show_rownames = F,
                     annotation_row = cell_metadata,
                     cluster_rows = T)
```



<br><br>
More example plots will be updated.  

<br><br>

